### PR TITLE
Fixing nullability annotations on Util.escape

### PR DESCRIPTION
**Pull request by jglick**
Two annotations were added to this method, both wrong, at least if we wish to keep its original behavior. Was causing me grief from FindBugs.

### Proposed changelog entries

N/A

### Desired reviewers

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/Util.java [hunk 1]
```diff
@@ -1094,8 +1094,8 @@ public static String singleQuote(String s) {
     /**
      * Escapes HTML unsafe characters like &lt;, &amp; to the respective character entities.
      */
-    @Nonnull
-    public static String escape(@Nonnull String text) {
+    @Nullable
+    public static String escape(@CheckForNull String text) {
```
Comment (stephenc) [inline core/src/main/java/hudson/Util.java]: The check for null should propagate, either the parameter is nullable in which case the return type is too (and nullable is the package default) or the return type should be check for null (which it shouldn’t imho)

Comment (jglick) [reply core/src/main/java/hudson/Util.java]: I am not sure what you are proposing, exactly.

The existing behavior, which we cannot compatibly change, is that the return value will be null iff the parameter is null. So, the parameter is marked `@CheckForNull` as a reminder inside the method that we do, in fact, need to check for null, and a note to callers that passing null is permitted; and the return value is `@Nullable` to emphasize to callers that a null check may or may not be needed, depending on what you know about the parameter.

**Review by Wadeck — APPROVED** [review]

**Comment by jglick** [comment]
@reviewbybees done

**Comment by oleg-nenashev** [comment]
I do not recall why I did that in https://github.com/jenkinsci/jenkins/pull/1223. Probably the idea was to avoid usage of null strings, but I agree it makes sense to change the annotation.

@stephenc Are you satisfied by the response?

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**